cpp/ql/src/Security/CWE/CWE-611/XXE.ql

```diff
@@ -188,6 +188,42 @@ class CreateLSParser extends Function {
   }
 }
 
+/**
+ * A call to a `libxml2` function that parses XML.
+ */
+class Libxml2ParseCall extends FunctionCall {
+  int optionsArg;
+
+  Libxml2ParseCall() {
+    exists(string fname | this.getTarget().getName() = fname |
+      fname = ["xmlCtxtUseOptions"] and optionsArg = 1
+      or
+      fname = ["xmlReadFile"] and optionsArg = 2
+      or
+      fname = ["xmlCtxtReadFile", "xmlParseInNodeContext", "xmlReadDoc", "xmlReadFd"] and
+      optionsArg = 3
+      or
+      fname = ["xmlCtxtReadDoc", "xmlCtxtReadFd", "xmlReadMemory"] and optionsArg = 4
+      or
+      fname = ["xmlCtxtReadMemory", "xmlReadIO"] and optionsArg = 5
+      or
+      fname = ["xmlCtxtReadIO"] and optionsArg = 6
+    )
+  }
+
+  /**
+   * Gets the argument that specifies `xmlParserOption`s.
+   */
+  Expr getOptions() { result = this.getArgument(optionsArg) }
+}
+
+/**
+ * An `xmlParserOption` for `libxml2` that is considered unsafe.
+ */
+class Libxml2BadOption extends EnumConstant {
+  Libxml2BadOption() { this.getName().matches(["XML_PARSE_NOENT", "XML_PARSE_DTDLOAD"]) }
```

Comment:
`_` is interpreted as "any character" by `matches`. I'm guessing that's not what we want here? As far as I can see, we really just want:
```ql
this.getName() = ["XML_PARSE_NOENT", "XML_PARSE_DTDLOAD"]
```
don't we?

Reply:
Yes, well spotted, and I'm not sure why I wrote `matches` there in the first place!

Reply:
Updated.